lib/chef/provisioning/docker_driver/driver.rb

```diff
@@ -176,7 +176,7 @@ def destroy_machine(action_handler, machine_spec, machine_options)
       Chef::Log.debug("Removing #{container_name}")
       container.delete
 
-      if !machine_spec.attrs[:keep_image] && !machine_options[:keep_image]
+      unless machine_spec.attrs[:keep_image] || machine_options[:keep_image]
```

Comment:
I believe, we need to remove this block completely. We have separate method for image deletion (#51).